internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -441,6 +443,12 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 			}
 		}
 
+		// Validate the name can be used as a kubernetes resource name by checking if the name
+		// is valid RFC 1123 subdomain.
+		if errs := validation.IsDNS1123Subdomain(cd.GetName()); len(errs) > 0 {
```

Comment:
This is too strict, subdomain validation is only needed for components that have names that end up as part of the internal dns structures. Resources like ... just about every CRD kind, and service accounts can have invalid subdomain chars like `:` or `.` and still be valid kubernetes names.

Reply:
similar fix to the same issue: https://github.com/kubernetes-sigs/cli-utils/pull/221